rocky/rocky/views/ooi_view.py

```diff
@@ -61,20 +56,42 @@ def get_ooi_profile_types(self) -> Set[ScanProfileType]:
             return self.scan_profile_types
         return {ScanProfileType(s) for s in self.clearance_type}
 
+    def get_ooi_types(self) -> Set[Type[OOI]]:
+        if not self.filtered_ooi_types:
+            return self.ooi_types
+        return {type_by_name(t) for t in self.filtered_ooi_types if t not in _EXCLUDED_OOI_TYPES}
+
     def get_queryset(self) -> OOIList:
-        return self.get_list(
-            observed_at=self.get_observed_at(),
+        if not (self.filtered_ooi_types or self.clearance_level or self.clearance_type):
+            return self.object_list
+
+        # A filtered OOI list
+        return OOIList(
+            self.octopoes_api_connector,
+            ooi_types=self.get_ooi_types(),
+            valid_time=self.get_observed_at(),
             scan_level=self.get_ooi_scan_levels(),
             scan_profile_type=self.get_ooi_profile_types(),
         )
 
+    def get_all_oois_and_count(self) -> Tuple[int, OOIList]:
```

Comment:
I think we can just return `all_oois` here and do `len()` in the `setup` method. Though I was wondering: why can't we use the `get_queryset` method for that? Since we are using the `setup` method I suppose a lot of the used properties might not have been set versus the way we retrieve them in `get_queryset`, but I do not have the Django View execution order fresh in mind at this point.

Reply:
Yes, the Django way, the setup method is called before get_queryset, we can off course move everything into the get_queryset, it was more of the way we use setup, to initiate things there only.

Reply:
I have use `setup` here foremost of how the information is distributed across the View. The filtered items will also be shown as text and also for `get_context` to use in templates

Reply:
> Yes, the Django way, the setup method is called before `get_queryset`, we can off course move everything into the `get_queryset`, it was more of the way we use setup, to initiate things there only.